Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM gentoo/portage:latest as portage
+FROM gentoo/stage3:latest as gentoo
+ENV ARCH=mips CROSS_COMPILE=mipsr5900el-unknown-linux-gnu-
+
+COPY --from=portage /var/db/repos/gentoo /var/db/repos/gentoo
+
+# qemu, croosdev and clean
+COPY crossdev.conf /etc/portage/repos.conf/
+RUN \
+	mkdir -p /var/db/repos/localrepo-crossdev/{profiles,metadata} && \
+	echo 'crossdev' > /var/db/repos/localrepo-crossdev/profiles/repo_name && \
+	echo 'masters = gentoo' > /var/db/repos/localrepo-crossdev/metadata/layout.conf && \
+	chown -R portage:portage /var/db/repos/localrepo-crossdev && \
+	emerge bc crossdev dev-vcs/git && \
+	crossdev --stage4 --target mipsr5900el-unknown-linux-gnu && \
+	mkdir -p initramfs/ps2/{lib/firmware/ps2,bin,dev,etc,mnt,proc,root,sbin,sys,tmp,usr,usr/bin,usr/sbin,var} && \
+	emerge -v app-eselect/eselect-repository && \
+	eselect repository add frno7 git https://github.com/frno7/gentoo.overlay && \
+	emaint sync -r frno7 && \
+	emerge -v dev-libs/glib && \
+	emerge -v app-emulation/qemu-mipsr5900el && \
+	rm /var/cache/distfiles/*
+# busybox
+RUN \
+	USE="prefix-guest static" emerge-mipsr5900el-unknown-linux-gnu -v sys-apps/busybox
+# iopmod
+RUN \
+	git clone https://github.com/frno7/iopmod --depth 1 && \
+	cd iopmod && make
+	# make clean && git checkout gamepad && make && \
+	# cp module/*.irx ../initramfs/ps2/lib/firmware/ps2/
+# initramfs
+COPY initramfs/ps2/init ./initramfs/ps2/
```

Comment:
Is `./initramfs/ps2/` in the root directory of the system? Maybe we could collect and organise the generated files in [`/srv`](https://tldp.org/LDP/Linux-Filesystem-Hierarchy/html/srv.html), which normally contains site-specific data which is served by this system?

Reply:
Currently everything is located under / (as mentioned in readme), of course some repos can be populated inside some directory.

Reply:
Also I always a bit afraid of using relative paths inside docker container, so I am sticking with absolute paths.

Reply:
> Also I always a bit afraid of using relative paths inside docker container, so I am sticking with absolute paths.

`./initramfs/ps2/` is actually a relative path (if you do `ls -a` you’ll find `.` and `..` in every directory), so `COPY initramfs/ps2/init ./initramfs/ps2/` looks like a copy to itself.

Nevertheless, I propose that we use `/srv/initramfs/ps2/`, and likewise with the other generated (“served”) files.

Reply:
Added **WORKDIR** `/srv`

Reply:
> Added **WORKDIR** `/srv`

Hmm. I saw in the [WORKDIR](https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#workdir) documentation that

> It's recommended to not use the `WORKDIR` instruction in your `Dockerfile`.

but I’m not sure if there are any alternatives. Also,

> `COPY initramfs/ps2/init ./initramfs/ps2/`

looks like it would be the same as

> `COPY initramfs/ps2/init initramfs/ps2/`

since `./` does mean the _current_ destination directory for `COPY`, doesn’t it? To avoid the `WORKDIR` directive, as its documentation recommends, maybe it could be `COPY initramfs/ps2/init /srv/initramfs/ps2/` instead?

Other than that, are you ready to merge this? I looked at your fork [mipsr5900el-gentoo-linux-gnu](https://github.com/AKuHAK/mipsr5900el-gentoo-linux-gnu) and I guess that packages need to be made ‘public’ to (as per the fourth item in https://github.com/frno7/linux/pull/69#issue-1190551260) to view the result of the action? That is, to inspect the generated IRX files, etc.?

Reply:
> I saw in the WORKDIR documentation that

It is a different [WORKDIR](https://docs.docker.com/engine/reference/builder/#workdir). WORKDIR inside Docker container is a common practice and wildly used.

Yes, currently ir is ready for merge. It can be made better in the future, but for practical use it is currently ready for use.

You can make package public only after it will be produced for the first time, so only after an hour after merging.

Reply:
> Hmm. I saw in the [WORKDIR](https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#workdir) documentation that

You are correct, WORKDIR needs to be redefined, cause it can lead to some problems. Fixed it in the same way as in [official Gentoo image](https://gitweb.gentoo.org/proj/docker-images.git/tree/stage3.Dockerfile) that is used as the base.

Reply:
> Other than that, are you ready to merge this?

Oh, I will check the last thing. That Github Action can sign Docker image, I will check if it works for us, just nice addition. Other than that, everything other is fine and ready for merge.


> I guess that packages need to be made ‘public’ to (as per the fourth item in [frno7/linux#69 (comment)](https://github.com/frno7/linux/pull/69#issue-1190551260))

It is possible to change package settings only after the first package is built. So, you need to merge this PR and then you need to wait until Github Action will produce the first package. Only after that, you will be able to change its visibility.